lib/webdriveragent.js

```diff
@@ -287,6 +288,73 @@ class WebDriverAgent {
     }
   }
 
+
+  /**
+   * @typedef {Object} LaunchingEnvironmentOptions
+   * @property {number} USE_PORT
+   * @property {string} WDA_PRODUCT_BUNDLE_IDENTIFIER
+   */
+
+  /**
+   * Launch WDA with preinstalled package with 'xcrun devicectl device process launch'.
+   * The WDA package must be prepared properly like published via
+   * https://github.com/appium/WebDriverAgent/releases
+   * with proper sign for this case.
+   *
+   * When we implement launching XCTest service via appium-ios-device,
+   * this implementation can be replaced with it.
+   *
+   * @param {LaunchingEnvironmentOptions}  launchingEnv launching environment for WebDriverAgent.
+   * @return {Promise<void>}
+   */
+
+  async _launchWdaViaDeviceCtl(launchingEnv) {
+    // FIXME: use appium-xcuitest-driver's Devicectl. Maybe it needs to be included in the `this.device`?
+
+    const envVar = {
+      // the port number must be string for the devicectl command.
+      USE_PORT: `${launchingEnv.USE_PORT}`,
+      WDA_PRODUCT_BUNDLE_IDENTIFIER: launchingEnv.WDA_PRODUCT_BUNDLE_IDENTIFIER
+    };
+
+    let xcrunBinnaryPath;
+    try {
+      xcrunBinnaryPath = await fs.which('xcrun');
+    } catch (e) {
+      throw new Error(
+        `xcrun has not been found in PATH. ` +
+          `Please make sure XCode development tools are installed`,
+      );
+    }
+
+    const {stdout} = await exec(xcrunBinnaryPath, [
```

Comment:
maybe we could move the devicectl class from xcuitest driver to this module to avoid extra repetition

Reply:
Yea, I'm thinking of doing that as a separate follow-up pr